presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkLocalShuffleWriteInfo.java

```diff
@@ -30,14 +30,22 @@
         implements PrestoSparkShuffleWriteInfo
 {
     private final int numPartitions;
+    // Unique identifier for each query generated by Presto.
+    private final String queryId;
```

Comment:
I am thinking it would be safer to use `StageExecutionId` instead of `QueryId`. `StageExecutionId` includes the queryid and additionally includes the `attemptNum` so it would be robust again scenarios where the shuffle write fails midway and the stage is re-executed ?